internal/rukpak/util/hash_test.go

```diff
@@ -0,0 +1,76 @@
+package util_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/operator-framework/operator-controller/internal/rukpak/util"
+)
+
+func TestDeepHashObject(t *testing.T) {
+	tests := []struct {
+		name         string
+		wantErr      bool
+		obj          interface{}
+		expectedHash string
+	}{
+		{
+			name:    "populated obj",
+			wantErr: false,
+			obj: struct {
+				str string
+				num int
+				obj interface{}
+				arr []int
+				b   bool
+				n   interface{}
+			}{
+				str: "foobar",
+				num: 22,
+				obj: struct{ foo string }{foo: "bar"},
+				arr: []int{0, 1},
+				b:   true,
+				n:   nil,
+			},
+			expectedHash: "16jfjhihxbzhfhs1k5mimq740kvioi98pfbea9q6qtf9",
+		},
+		{
+			// An empty object generates the same hash as the populated object above... that feels wrong?
```

Comment:
Commenting here to call attention to this.

Reply:
Figured this out thanks to @joelanford , the object passed into `DeepHashObject` needs to have exported fields or else the JSON encoder will treat the object the same as an entirely empty object. I've updated the tests to show this behavior.